Moved Table.explode to IR.

cseed opened the pull request.

jbloom22 commented on src/main/scala/is/hail/expr/Relational.scala:
```diff
@@ -1131,3 +1131,59 @@ case class TableMapGlobals(child: TableIR, newRow: IR) extends TableIR {
+  def children: IndexedSeq[BaseIR] = Array(child)
+
+  private val columnIdx = child.typ.rowType.fieldIdx(column)
+  private val columnType = child.typ.rowType.types(columnIdx).asInstanceOf[TContainer]
```
check that columnType is a container type in Python?

cseed replied:
Good catch, I thought that was happening.

jbloom22 commented on src/main/scala/is/hail/expr/Relational.scala:
```diff
@@ -1131,3 +1131,59 @@ case class TableMapGlobals(child: TableIR, newRow: IR) extends TableIR {
+            val n = lengthF()(rv.region, rv.offset, false)
+            Iterator.range(0, n)
+              .map { i =>
+                // rv.region.clear(end)
```
commented code

cseed commented:
I changed is_container to match the meaning in Scala (is array, set or dict) and added is_compound to have the old meaning (type that can contain other types).
